src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruitName;
+    private int transactionQuantity;
+    public FruitTransaction(Operation operation, String fruitName, int transactionQuantity) {
+        this.operation = operation;
+        this.fruitName = fruitName;
+        this.transactionQuantity = transactionQuantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+    }
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public static Operation convertFromCode(String code) {
```

Comment:
The `convertFromCode` method should be placed in a more appropriate class, such as a utility class or within the Operation enum itself, to follow the SOLID principles, specifically the Single Responsibility Principle. It's not the responsibility of `FruitTransaction` to know how to convert operation codes.

Reply:
Please follow the suggestion from ai-mentor. Place it in Enum.